vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -231,17 +268,24 @@ def _validate_column_type(self, targeted_data: pd.DataFrame) -> Literal["numeric
             )
 
     @staticmethod
-    def _get_min_max(targeted_data: pd.DataFrame) -> tuple[float, float]:
+    def _get_min_max(targeted_data: pd.DataFrame, current_value=None) -> tuple[float, float]:
+        targeted_data = pd.concat([targeted_data, pd.Series(current_value)]).stack().dropna()  # noqa: PD013
+
+        _min = targeted_data.min(axis=None)
+        _max = targeted_data.max(axis=None)
+
         # Use item() to convert to convert scalar from numpy to Python type. This isn't needed during pre_build because
         # pydantic will coerce the type, but it is necessary in __call__ where we don't update model field values
         # and instead just pass straight to the Dash component.
-        return targeted_data.min(axis=None).item(), targeted_data.max(axis=None).item()
+        # However, in some cases _min and _max are already Python types and so item() call is not required.
+        _min = _min if not hasattr(_min, "item") else _min.item()
+        _max = _max if not hasattr(_max, "item") else _max.item()
+
+        return _min, _max
 
     @staticmethod
-    def _get_options(targeted_data: pd.DataFrame) -> list[Any]:
-        # Use tolist() to convert to convert scalar from numpy to Python type. This isn't needed during pre_build
-        # because pydantic will coerce the type, but it is necessary in __call__ where we don't update model field
-        # values and instead just pass straight to the Dash component.
+    def _get_options(targeted_data: pd.DataFrame, current_value=None) -> list[Any]:
         # The dropna() isn't strictly required here but will be in future pandas versions when the behavior of stack
         # changes. See https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#whatsnew-210-enhancements-new-stack.
-        return np.unique(targeted_data.stack().dropna()).tolist()  # noqa: PD013
+        targeted_data = pd.concat([targeted_data, pd.Series(current_value)]).stack().dropna()  # noqa: PD013
```

Comment:
Sorry, but what do you exactly mean by "this"?
It's true that the `targeted_data` can potentially have multiple columns which is the data from several underlying datasets. Even though it's a single column pd.DataFrame, we found it practical to add the `current_value` as the additional column there and to find min/max/options then.

Reply:
Not sure now whether its single column or mutiple column :) But nevermind. I mean with "this", why there was such a complicated operation if really we want a set of unique values. But suspect it is because we first need to concatenate all the columns, and the current value before we can do the set operation

Reply:
I had many different versions of this method and the latest solution became a winner as the shortest, and as the most robust way to find unique values.